src/Processors/Formats/Impl/NativeORCBlockInputFormat.cpp

```diff
@@ -728,14 +906,35 @@ void NativeORCBlockInputFormat::prepareFileReader()
         format_settings.orc.case_insensitive_column_matching);
 
     const bool ignore_case = format_settings.orc.case_insensitive_column_matching;
-    std::unordered_set<String> nested_table_names = Nested::getAllTableNames(getPort().getHeader(), ignore_case);
-
-    for (size_t i = 0; i < schema.columns(); ++i)
+    const auto & header = getPort().getHeader();
+    const auto & file_schema = file_reader->getType();
+    std::unordered_set<UInt64> include_typeids;
+    for (const auto & column : header)
```

Comment:
So, now we don't need `schema` that we got from `getFileReaderAndSchema`. Let's remove `getFileReaderAndSchema` function with `getFileReader`, and create schema from orc type in `NativeORCSchemaReader::readSchema`